src/index.js

```diff
@@ -29,36 +26,56 @@ class Resource {
    * {
    *   id: '5869485767494'
    *   params: {
-   *     limit: 5
+   *     limit: 5,
+   *     since: 1467964293258
    *   }
    * }
    *
    * @param query
    * @returns {Promise}
    */
-  get (query) {
-    if (!query) {
-      query = {};
-    }
+
+  get(query, results) {
+    var query = query || {}
+    var results = results || [];
+    var that = this;
+    var deferred = Promise.defer();
 
     this.signatureFactory.setUrl(this.url);
     this.signatureFactory.handleQuery(query);
     const signature = this.signatureFactory.sign();
 
-    const options = {
-      url: this.getBaseUrl() + signature.url,
+    var requestOptions = {
+      protocol: 'https:',
```

Comment:
The config has a protocol defined `protocol: 'https',`, why do you need to define it here as well?